### PR TITLE
Fix docker registry error on Nexus and coding.net Registry

### LINKED ISSUE #188: Nexus 和 Coding.net Registry 不支持的bug
### 运行环境
* `应用版本`:
     * `atomci-frontend`: latest
     * `atomci`:latest
* `Kubernetes`:v1.23

### 问题描述
1. Nexus作为Docker Registry 添加不上
原因是：Nexus的认证参数是非规范的
88290c6cc7dc6e4130c9c4371bc7eed

2. Coding.net的Docker制品库 添加不上
原因是：认证的realm中有特殊的“=”符号，导致字符截取错误
f214869b9b0d912eb94c10aae28e5585

### PR DESCRIPTION
**Pull request by sampsonye**
### Which changes (Bug/Feature):

Fixes #188 

### Special notes for reviewers:

```
使用strings.EqualFold进行忽略大小写的比较
使用strings.index来进行字符串判断，防止截取意外的“=”
```

### CODE REVIEW

File: internal/core/settings/registryhub.go [hunk 1]
```diff
@@ -33,19 +33,23 @@ func TryLoginRegistry(basicUrl, username, password string, insecure bool) error
 
 	//get Auth Info
 	auth := resp.Header.Get("Www-Authenticate")
-	if strings.HasPrefix(auth, "Basic") {
+
+	if strings.EqualFold("Basic", auth[:5]) {
 		authBaseUrl = url
 		queryParams = nil
-	} else if strings.HasPrefix(auth, "Bearer") {
+	} else if strings.EqualFold("Bearer", auth[:6]) {
```
Comment (colynn) [inline internal/core/settings/registryhub.go:40]: 看到边界点很多，建议添加下单元测试

**Review by colynn — APPROVED** [review]
/lgtm

File: internal/core/settings/registryhub.go (hunk 1 above)
Comment (sampsonye) [reply internal/core/settings/registryhub.go:40]: done

**Review by colynn — APPROVED** [review]
np

**Pull request merged**